lib/active_model/serializer/adapter/fragment_cache.rb

```diff
@@ -41,31 +33,38 @@ def fetch
 
         private
 
-        # Given a serializer class and a hash of its cached and non-cached serializers
+        def serialize(object, serializer_class)
+          SerializableResource.new(
+            object,
+            serializer: serializer_class,
+            adapter: adapter.class
+          ).serializable_hash
+        end
+
+        # Given a hash of its cached and non-cached serializers
         # 1. Determine cached attributes from serializer class options
         # 2. Add cached attributes to cached Serializer
         # 3. Add non-cached attributes to non-cached Serializer
-        def cached_attributes(klass, serializers)
-          attributes            = serializer.class._attributes
-          cached_attributes     = (klass._cache_only) ? klass._cache_only : attributes.reject { |attr| klass._cache_except.include?(attr) }
+        def cached_attributes(serializers)
+          klass                 = serializer.class
+          attributes            = klass._attributes
+          cache_only            = klass._cache_only
+          cached_attributes     = cache_only ? cache_only : attributes - klass._cache_except
           non_cached_attributes = attributes - cached_attributes
+          attributes_keys       = klass._attributes_keys
 
-          cached_attributes.each do |attribute|
-            options = serializer.class._attributes_keys[attribute]
-            options ||= {}
-            # Add cached attributes to cached Serializer
-            serializers[:cached].constantize.attribute(attribute, options)
-          end
+          add_attributes(serializers[:cached], cached_attributes, attributes_keys)
+          add_attributes(serializers[:non_cached], non_cached_attributes, attributes_keys)
+        end
 
-          non_cached_attributes.each do |attribute|
-            options = serializer.class._attributes_keys[attribute]
-            options ||= {}
-            # Add non-cached attributes to non-cached Serializer
-            serializers[:non_cached].constantize.attribute(attribute, options)
+        def add_attributes(serializer, attributes, attributes_keys)
+          attributes.each do |attribute|
+            options = attributes_keys[attribute] || {}
+            serializer.attribute(attribute, options)
```

Comment:
this is somewhat pre-existing, but may be we can make it more clear this is the `fragment_serializer_class` that we're building up in an atypical way?

Instead of what we have now

```ruby
klass      = serializer.class
cached     = "#{to_valid_const_name(name)}CachedSerializer"
cached_serializer =  Object.const_set(name, Class.new(ActiveModel::Serializer))
cached_attributes.each do |attribute|
  cached_serializer.attribute(attribute, klass._attribute_keys[attribute] || {})
end
```

we'd have

```ruby
klass      = serializer.class
cached     = "#{to_valid_const_name(name)}CachedSerializer"
cached_serializer =  Object.const_set(name, Class.new(ActiveModel::Serializer) do
  cached_attributes.each do |attribute_name|
    attribute attribute_name, (klass._attribute_keys[attribute_name] || {})
  end
end)
```